src/darwin/Framework/CHIPTests/TestHelpers/MTRMockCB.m

```diff
@@ -0,0 +1,690 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTRMockCB.h"
+
+#import "MTRDefines_Internal.h"
+
+#import <XCTest/XCTest.h>
+#import <objc/runtime.h>
+#import <os/log.h>
+#import <stdatomic.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MTRMockCBPeripheralDetails : NSObject
+
+- (instancetype)initWithIdentifier:(NSUUID *)identifier;
+
+@property (readonly, nonatomic, weak) MTRMockCB * mock;
+@property (readonly, nonatomic, strong) NSUUID * identifier;
+@property (readonly, nonatomic, copy) NSString * name;
+@property (readonly, nonatomic, copy) NSDictionary<NSString *, id> * advertisementData;
+
+@property (nonatomic, assign) CBPeripheralState state;
+@property (nonatomic, nullable, copy) NSDictionary<NSString *, id> * extraAdvertisementData;
+@property (nonatomic, copy) NSArray<CBUUID *> * services;
+
+@end
+
+@interface MTRMockCBCentralManager : NSObject
+
+- (instancetype)_initWithMock:(MTRMockCB *)mock;
+- (void)_didUpdateState;
+- (void)_maybeDiscoverPeripheral:(MTRMockCBPeripheralDetails *)details;
+
+// MARK: CBManager
+
+@property (nonatomic, assign, readonly) CBManagerState state;
+@property (nonatomic, assign, readonly) CBManagerAuthorization authorization;
+@property (class, nonatomic, assign, readonly) CBManagerAuthorization authorization;
+
+// MARK: CBCentralManager
+
+@property (nonatomic, weak, nullable) id<CBCentralManagerDelegate> delegate;
+@property (nonatomic, assign, readonly) BOOL isScanning;
+
++ (BOOL)supportsFeatures:(CBCentralManagerFeature)features;
+
+- (instancetype)init;
+- (instancetype)initWithDelegate:(nullable id<CBCentralManagerDelegate>)delegate
+                           queue:(nullable dispatch_queue_t)queue;
+- (instancetype)initWithDelegate:(nullable id<CBCentralManagerDelegate>)delegate
+                           queue:(nullable dispatch_queue_t)queue
+                         options:(nullable NSDictionary<NSString *, id> *)options;
+
+- (NSArray<CBPeripheral *> *)retrievePeripheralsWithIdentifiers:(NSArray<NSUUID *> *)identifiers;
+- (NSArray<CBPeripheral *> *)retrieveConnectedPeripheralsWithServices:(NSArray<CBUUID *> *)serviceUUIDs;
+- (void)scanForPeripheralsWithServices:(nullable NSArray<CBUUID *> *)serviceUUIDs options:(nullable NSDictionary<NSString *, id> *)options;
+- (void)stopScan;
+- (void)connectPeripheral:(CBPeripheral *)peripheral options:(nullable NSDictionary<NSString *, id> *)options;
+- (void)cancelPeripheralConnection:(CBPeripheral *)peripheral;
+- (void)registerForConnectionEventsWithOptions:(nullable NSDictionary<CBConnectionEventMatchingOption, id> *)options;
+
+@end
+
+@interface MTRMockCBPeripheral : NSObject <NSCopying>
+
+- (instancetype)_initWithDetails:(MTRMockCBPeripheralDetails *)details manager:(MTRMockCBCentralManager *)manager;
+
+@property (readonly, strong, nonatomic) MTRMockCBCentralManager * manager; // not API, but used by BlePlatformDelegateImpl via KVC
```

Comment:
Not sure what KVC is here...

Reply:
Key Value Coding, it's another of those ObjC duck typing / reflection things... Our BLE Platform Manager does this to access this internal property:
```
CBCentralManager * manager = (CBCentralManager *) [peripheral valueForKey:@"manager"];
```